### PR TITLE
Add kosp2e asr recipe (Bootcamp Project)

### PR DESCRIPTION
**Pull request by zheedong**
## What did you change?

Added a new ASR recipe for the kosp2e dataset under egs2/kosp2e/asr1.

This includes:
* Complete data preparation, training, and decoding pipeline scripts (run.sh, setup_data.py, etc.)
* Model configuration files and local utilities
* A detailed README.md describing dataset setup, training instructions, and results
* Example Conformer-BPE2000 baseline experiment results

---

## Why did you make this change?

To add official ESPnet2 support for the kosp2e (Korean Speech Perception and Production Experiment) dataset —
a high-quality Korean speech corpus useful for ASR and perception modeling research.

---

## Is your PR small enough?

Yes.
The PR adds one new recipe directory (egs2/kosp2e/asr1) and does not modify ESPnet core components.
22 files changes, but 587 lines added. I'm not sure I need to divide this PR.

---

## Additional Context

**Environment**
- Python 3.10.19 / PyTorch 2.9.0+cu128 / ESPnet 202509
- Model: Conformer (BPE=2000)
- Decode: Transformer LM (valid.acc.ave)

**Results (Test Set)**

| Metric | Corr | Sub | Del | Ins | Err | S.Err |
|:--|--:|--:|--:|--:|--:|--:|
| **WER** | 77.1 | 20.4 | 2.6 | 4.4 | 27.4 | 76.4 |
| **CER** | 92.5 | 5.7 | 1.8 | 1.7 | 9.2 | 76.4 |
| **TER** | 89.4 | 8.6 | 2.0 | 2.1 | 12.7 | 76.4 |

**References**
- kosp2e paper: https://arxiv.org/abs/2107.02875

### CODE REVIEW

File: egs2/kosp2e/asr1/run.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+./asr.sh \
+    --nj 8 \
+    --inference_nj 8 \
```
Comment (ftshijt) [inline egs2/kosp2e/asr1/run.sh]: usually, we would set these as default by removing the setup in run.sh

File: egs2/kosp2e/asr1/run.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    --asr_config conf/train_asr_conformer.yaml \
+    --lm_config conf/train_lm_transformer.yaml \
+    --inference_config conf/decode_asr.yaml \
+    --stage 2 \
+    --train_set train \
+    --valid_set val \
+    --test_sets "test" \
+    --nbpe 2000 \
+    --ngpu 4 \
+    --batch_size 1 \
```
Comment (ftshijt) [inline egs2/kosp2e/asr1/run.sh]: To set batch size as 1 is a bit strange. May I if you can validate the setting?

File: egs2/kosp2e/asr1/setup.sh [hunk 3]
```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
```
Comment (ftshijt) [inline egs2/kosp2e/asr1/setup.sh]: This setup is for template purpose, so we can safely remove it

**Review by ftshijt — COMMENTED** [review]
Apart from the gemini suggestions, please also address the following

File: egs2/kosp2e/asr1/README.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+```
+./run.sh
+```
+
+This script runs the standard ESPnet2 training pipeline with the Conformer model and BPE-2000 tokenization.
+
+---
+
+# 4. Results
```
Comment (sw005320) [inline egs2/kosp2e/asr1/README.md]: Can you upload a model in HF and add a link?
Please also check the other recipe.

File: egs2/kosp2e/asr1/run.sh (hunk 2 above)
Comment (sw005320) [reply egs2/kosp2e/asr1/run.sh]: Also, we should not use multiple GPUs for this scale of the recipe.
Please refer to https://github.com/espnet/espnet/blob/master/egs2/librispeech_100/asr1/run.sh

File: egs2/kosp2e/asr1/setup_data.py
Comment (sw005320) [inline egs2/kosp2e/asr1/setup_data.py:1]: This should move to `/local`

**Comment by sw005320** [comment]
Thanks, @zheedong!
